### ISSUE TITLE
 Refactor PNI "callout-box" version of newsletter signup form

### ISSUE THREAD

**Issue by mmmavis** [MozillaFoundation/foundation.mozilla.org#11578]
This is part of the refactoring work for newsletter signup form (epic #11225)

For the PNI "callout-box" version of newsletter signup form, extract the existing code from `join.jsx`, and create a similar component based on what we did for [the blog body streamfield block implementation](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/source/js/components/newsletter-signup/organisms/blog-body-signup.jsx).

For detailed audit report, see [this sheet](https://docs.google.com/spreadsheets/d/1EeWshn5N7f6eEZ8eBdMS_IEF1RtFuaXFeUrwLEX96js/edit#gid=0). Look for the "PNI (newsletter box)" column

Example: https://foundation.mozilla.org/en/privacynotincluded/euki/

**Comment by mmmavis** [comment]
Hi @sabrinang @nancyt1 , (tagging you both as I remember you worked on form fields for the design system)

Quick question for you - I'm wondering if we want to keep the existing form style for the PNI callout form or is it okay that we update it so it matches the other signup forms. I've attached screenshots below for comparison. It's totally fine if we can't make a decision now, I can refactor the code and keep the existing style. Just let me know! :) Thanks!

The biggest difference on I found are:

- form fields and submit button has no gap in between (vs with gap)
- email field is taller to match the submit button height (vs normal form field height)
- form field border is light gray (vs black)

# Current

<img width="762" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/2896608/d1a47dc9-0c04-4110-939e-892111e1189b">

<img width="772" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/2896608/29832f6f-1e41-4187-b3e0-4031e301a042">


---

# If replacing using the new component (without adding further styling changes in code)


<img width="785" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/2896608/57e60e6b-fe47-4af5-8942-2992650d6316">

<img width="781" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/2896608/351fd5dc-b4ac-437c-8d06-26f61113ad86">

**Comment by sabrinang** [comment]
For context, the PNI components have been separate from the Foundation design system work done earlier this year and were made separate. Now that the foundation design system is more defined, PNI styles/components could be checked to see what could be aligned or expanded foundation design system or what is deliberately unique that belongs in it's sub brand or tier tokens set.

I will lean into @nancyt1 thoughts too as she originally designed the PNI newsletter component but some of my thought are:

- I believe h3 would work for this but if PNI is using a font size between h2-h3 often enough use cases across the site we'd consider adding it for PNI
- Looking at the other newsletter styles and forms, buttons either have a gap or on a separate line. I’d lean towards the gap (use a space token)
- in the figma files of the design system, the default input field height (40px) matches buttons height (40px) but I think this is yet to be reflected in code but having them be the same height would be ideal
- The “filled” style of input fields have a grey boarder _(vs. “Outlined” style which has black so we’d use “filled” here)_

**Comment by nancyt1** [comment]
+1 to all Sabrina said! Let's consolidate and go with H3 for heading and using the "filled" styling for the form if possible. Ideally the input field would match the button height but I'm cool with matching existing forms if it makes it easier for us to update all of them at once in the future. Is that part of the goal of the refactor?

**Comment by mmmavis** [comment]
@sabrinang @nancyt1 Thanks for the feedback! I looked into changing the input box height and button height but realized the work will touch quite a bit of other things in the codebase (e.g., the "Sign up" button height isn't 40px, line height and padding also need fixing etc). 

To keep the work simple for this particular ticket, I will

- change heading type to `h3` 
- make sure to keep the "filled" styling
- add gap in between the input field and button (like the "blog-body" forms on the site)

Once we have refactored all the existing signup forms in the codebase, it should be a lot easier to introduce additional variations or styling fixes.
